x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvSliceBooleanEvaluator.java

```diff
@@ -0,0 +1,140 @@
+import org.elasticsearch.xpack.ql.InvalidArgumentException;
+import org.elasticsearch.xpack.ql.tree.Source;
+
+/**
+ * {@link EvalOperator.ExpressionEvaluator} implementation for {@link MvSlice}.
+ * This class is generated. Do not edit it.
+ */
+public final class MvSliceBooleanEvaluator implements EvalOperator.ExpressionEvaluator {
+  private final Warnings warnings;
+
+  private final EvalOperator.ExpressionEvaluator field;
+
+  private final EvalOperator.ExpressionEvaluator start;
+
+  private final EvalOperator.ExpressionEvaluator end;
+
+  private final DriverContext driverContext;
+
+  public MvSliceBooleanEvaluator(Source source, EvalOperator.ExpressionEvaluator field,
+      EvalOperator.ExpressionEvaluator start, EvalOperator.ExpressionEvaluator end,
+      DriverContext driverContext) {
+    this.warnings = new Warnings(source);
+    this.field = field;
+    this.start = start;
+    this.end = end;
+    this.driverContext = driverContext;
+  }
+
+  @Override
+  public Block eval(Page page) {
+    try (BooleanBlock fieldBlock = (BooleanBlock) field.eval(page)) {
+      try (IntBlock startBlock = (IntBlock) start.eval(page)) {
+        try (IntBlock endBlock = (IntBlock) end.eval(page)) {
+          return eval(page.getPositionCount(), fieldBlock, startBlock, endBlock);
+        }
+      }
+    }
+  }
+
+  public BooleanBlock eval(int positionCount, BooleanBlock fieldBlock, IntBlock startBlock,
+      IntBlock endBlock) {
+    try(BooleanBlock.Builder result = driverContext.blockFactory().newBooleanBlockBuilder(positionCount)) {
+      position: for (int p = 0; p < positionCount; p++) {
+        boolean allBlocksAreNulls = true;
+        if (!fieldBlock.isNull(p)) {
+          allBlocksAreNulls = false;
+        }
+        if (startBlock.isNull(p)) {
+          result.appendNull();
+          continue position;
+        }
+        if (startBlock.getValueCount(p) != 1) {
+          if (startBlock.getValueCount(p) > 1) {
+            warnings.registerException(new IllegalArgumentException("single-value function encountered multi-value"));
+          }
+          result.appendNull();
+          continue position;
+        }
+        if (endBlock.isNull(p)) {
+          result.appendNull();
+          continue position;
+        }
+        if (endBlock.getValueCount(p) != 1) {
+          if (endBlock.getValueCount(p) > 1) {
+            warnings.registerException(new IllegalArgumentException("single-value function encountered multi-value"));
+          }
+          result.appendNull();
+          continue position;
+        }
+        if (allBlocksAreNulls) {
+          result.appendNull();
+          continue position;
+        }
+        try {
+          MvSlice.process(result, p, fieldBlock, startBlock.getInt(startBlock.getFirstValueIndex(p)), endBlock.getInt(endBlock.getFirstValueIndex(p)));
+        } catch (InvalidArgumentException e) {
+          warnings.registerException(e);
+          result.appendNull();
```

Comment:
This could happen after a `builder.beginPositionEntry()` and a few `builder.append*()` (technically also after a `builder.endPositionEntry()`, even though it seems more unlikely); in this case `appendNull() is not what we want.
We'll need to revert the partial changes probably, but I'm not sure we have high level logic for this.
@nik9000 I need your expertise here, maybe we can add a method to revert pending changes?

Reply:
ooof. Your right. It's invalid to call `appendNull` after calling `beginPositionEntry` and before calling `endPositionEntry` - in other words, once you've started a position you are guaranteeing it be not-null. We could all support for this, I think - either in a special builder if there's a bunch of overhead, or in the main builder if there isn't.